components/raftstore/src/store/peer.rs

```diff
@@ -3604,10 +3610,25 @@ where
 
     fn apply_reads<T>(&mut self, ctx: &mut PollContext<EK, ER, T>, ready: &Ready) {
         let mut propose_time = None;
-        let states = ready.read_states().iter().map(|state| {
-            let read_index_ctx = ReadIndexContext::parse(state.request_ctx.as_slice()).unwrap();
-            (read_index_ctx.id, read_index_ctx.locked, state.index)
-        });
+        let states: Vec<_> = ready
+            .read_states()
+            .iter()
+            .map(|state| {
+                let read_index_ctx = ReadIndexContext::parse(state.request_ctx.as_slice()).unwrap();
+                if let Some(memory_lock) = read_index_ctx.memory_lock {
+                    // there are no pending proposals in leader
+                    let start_ts: u64 = memory_lock;
+                    if self.ready_to_handle_unsafe_replica_read(state.index)
+                        && self.read_progress.read_indx_safe_ts.load(Ordering::SeqCst) < start_ts
+                    {
+                        self.read_progress
+                            .read_indx_safe_ts
```

Comment:
```suggestion
                            .read_index_safe_ts
```
Please use `idx` or `index`.

Reply:
i will change it